### PR TITLE
fix(18194): fix trigger to close window when clicking "back to safety"

### PR DESCRIPTION
**Pull request by DDDDDanica**
**Fixes**: https://app.zenhub.com/workspaces/extension-client-team-63529dce65cbb100265a3842/issues/gh/metamask/metamask-extension/18194

 **Before**: 
- [user visits blocked page] -> [window.location.href = "warning page URL" in contentScript]  -> [warning page] -> [user clicks "Back to safety"] -> [ window.close() ] -> not working in Firefox 

**Trials**: 

- [user visits blocked page] -> [ location.href = "warning page URL" ]  -> [warning page] -> [user clicks "Back to safety"] -> [window.open('location', '_self', '');window.close();] -> not working for Firefox

- [user visits blocked page] -> [window.location.replace("warning page URL") ] -> [warning page] -> [user clicks "Back to safety"] -> [window.close() ] -> not working for Firefox 

- [user visits blocked page] -> [window.location.hrer("warning page URL") ] -> [warning page] -> [user clicks "Back to safety"] -> [open a new blank page] -> [document.visibilitystate === hidden, detect user has left previous warning page] -> [window.close() inactive page] -> not working for Firefox 

**Temporary solution:**
- [user visits blocked page] -> [ location.href = "warning page URL" ]  -> [warning page] -> [user clicks "Back to safety"] -> [ location.replace("about:blank") ] -> page stays open on blank tab, works for both browsers 

Proposed by @danjm :
**- [user visits blocked page] -> [ location.href = "warning page URL" ]  -> [warning page] -> [user clicks "Back to safety"] -> [ location.replace(MM extension expanded view)] -> page stays open on a specific route, works for both browsers** 

**After**: 

https://user-images.githubusercontent.com/12678455/228697448-b548cce1-efd8-4753-9b88-8d134569dc11.mov

### CODE REVIEW

**Comment by Gudahtt** [comment]
Huh, interesting solution! It seems that this is an attempt to circumvent browser protections against programmatically closing a window, by "opening" the window programmatically first before closing it. I have seen references to this approach not working on some browser versions, but I don't know how that affects the range of versions we support.

The e2e tests here are passing, but the test for this close button might not be sufficient. Right now that test is configured to ensure the length of history on the phishing warning page is exactly 1 so that the browser considers it as being opened programmatically. The reason this test didn't catch the bug in the extension is that today the extension redirects to this page, resulting in a history length of 2. We should try to ensure the test accurately captures this scenario, so that it can protect against future regressions.

So I have 2 suggestions overall: consider browser support, and ensure the test is updated to catch this problem.

**Comment by danjm** [comment]
with the change here, the tab is now left open, right? https://github.com/MetaMask/phishing-warning/pull/84/commits/fca89f3c061d3e0477e6f9164b33517f42d73699

File: tests/defaults.spec.ts [hunk 1]
```diff
@@ -55,20 +54,13 @@ test('does nothing when the user tries to bypass the warning', async ({
...
-  const baseURL = config.use?.baseURL;
-  await page.evaluate((url) => window.location.replace(url), `${baseURL}/`);
-
-  const onClose = page.waitForEvent('close');
+  await page.goto('/');
   await page.getByRole('button', { name: 'Back to safety' }).click();
-
-  await onClose;
+  await expect(page.isClosed()).toBe(false);
+  // the redirection to MM extension will be tested in MM Extension e2e test
```
Comment (Gudahtt) [inline tests/defaults.spec.ts:63]: We can check in this test that the extension has at least been sent the correct message. See here for an example of how to do this: https://github.com/MetaMask/phishing-warning/blob/cb973a59e56108c754eb6f8fe88ef2c8e9b408c8/tests/bypass.spec.ts#L23

**Review by danjm — CHANGES_REQUESTED** [review]
Looks good. Just Mark's test suggestion to address

File: tests/defaults.spec.ts (hunk 1 above)
Comment (DDDDDanica) [reply tests/defaults.spec.ts]: Thank you Mark, this is very helpful, and very easy to use ~

**Review by danjm — APPROVED** [review]
LGTM

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**